lib/internal/checkout.ts

```diff
@@ -0,0 +1,126 @@
+import { ApiResponse, JustifiRequest, RequestMethod } from "./http";
+
+export enum CheckoutStatus {
+  Completed = "completed",
+  Created= "created",
+}
+
+export interface Checkout {
+  id: string;
+  accountId: string;
+  platformAccountId: string;
+  paymentAmount: number;
+  paymentCurrency: string;
+  paymentDescription: string;
+  paymentMethods: any;
+  paymentMethodGroupId: string;
+  status: CheckoutStatus;
+  successfulPaymentId: string,
+  paymentSettings: any,
+  createdAt: string;
+  updatedAt: string;
+}
+
+export interface CreateCheckoutPayload {
+  amount: number;
+  currency?: string;
+  description?: string;
+  origin_url?: string;
+  paymentMethodGroupId?: string;
+}
+
+export interface CompleteCheckoutPayload {
+  paymentToken: string;
+}
+
+export interface CheckoutApi {
+  listCheckouts(subAccountId?: string): Promise<ApiResponse<Checkout[]>>;
+  getCheckout(id: string): Promise<ApiResponse<Checkout>>;
+  updateCheckout(
+    id: string,
+    amount?: number,
+    description?: string
+  ): Promise<ApiResponse<Checkout>>;
+  createCheckout(
+    payload: CreateCheckoutPayload,
+    subAccountId?: string
+  ): Promise<ApiResponse<Checkout>>;
+  completeCheckout(
+    id: string,
+    idempotencyKey: string,
+    payload: CompleteCheckoutPayload,
+    subAccountId?: string
```

Comment:
You can remove this arg too

Reply:
We should update the docs, it is not needed. A checkout is already associated with a sub account